lib/types/syncRule.js

```diff
@@ -0,0 +1,29 @@
+import Rule from './rule';
+
+/**
+ * Helper rule that handles callback calling for synchronous validation rules.
+ */
+export default class SyncRule extends Rule {
+    /**
+     * @override
+     */
+    validate(params, callback) {
+        const res = this.validateSync(params);
+        if (res instanceof Object) {
```

Comment:
micro-optimization: add `if (res === true)` to exit early for rules that pass it.